apps/wallets/smart-wallet/next/src/components/header.tsx

```diff
@@ -1,66 +1,83 @@
-
 export const Header = () => {
-    const { authedUser, signOut } = useAuth();
+    const { jwt, logout, wallet, isLoadingWallet } = useAuth();
     const router = useRouter();
     const { toast } = useToast();
 
+    const hasWalletAndJwt = jwt != null && wallet != null && !isLoadingWallet;
+
     const handleLogout = () => {
-        signOut();
+        logout();
         router.push("/");
     };
 
     return (
         <div className="flex justify-between p-6 items-center">
-            <div className="justify-center items-center flex ">
+            <Link href="/" className="justify-center items-center flex">
                 <div className="hidden sm:block">
                     <Logo />
                 </div>
                 <div className="block sm:hidden">
                     <MobileLogo />
                 </div>
-            </div>
-            {authedUser ? (
-                <div className="flex gap-5 items-center">
+            </Link>
+
+            {hasWalletAndJwt || isLoadingWallet ? (
+                <div className="flex gap-5">
                     <DropdownMenu>
-                        <DropdownMenuTrigger asChild>
-                            <div className="flex items-center bg-skeleton rounded-full px-4 py-2 gap-2 cursor-pointer text-secondary-foreground">
-                                <WalletMinimal className="h-4 w-4" />
-                                <Typography>
-                                    {mockWalletId.substring(0, 6) +
-                                        "..." +
-                                        mockWalletId.substring(mockWalletId.length - 3, mockWalletId.length)}
-                                </Typography>
+                        <DropdownMenuTrigger asChild disabled={!hasWalletAndJwt}>
+                            <div className="flex items-center gap-5 cursor-pointer">
+                                <div className="flex items-center min-w-[150px] bg-skeleton rounded-full px-4 py-2 gap-2 text-secondary-foreground">
+                                    <WalletMinimal className="h-4 w-4" />
+                                    <Typography>
+                                        {hasWalletAndJwt
+                                            ? wallet.address.substring(0, 6) +
+                                              "..." +
+                                              wallet.address.substring(wallet.address.length - 3, wallet.address.length)
+                                            : "Loading..."}
+                                    </Typography>
+                                </div>
+                                <Avatar className="h-9 w-9">
+                                    <AvatarImage alt="User Avatar" src="" />
+                                    <AvatarFallback className="bg-skeleton">
+                                        <User className="h-5 w-5" />
+                                    </AvatarFallback>
+                                </Avatar>
```

Comment:
double checking this is the same right color (same as the background)?

Reply:
it's the same color as the wallet button background

Reply:
![Screenshot from 2024-08-14 14-56-54](https://github.com/user-attachments/assets/553b7248-971f-4328-a30d-c6591c6c3abe)

Reply:
v/ nice